centrallix/objectsystem/obj_datatypes.c

```diff
@@ -1442,26 +1455,33 @@ objDataToMoney(int data_type, void* data_ptr, pMoneyType m)
 		    ptr++;
 		    }
 		intval = 0;
-		m->FractionPart = 0;
-		m->WholePart = 0;
-		intval = strtoul(ptr, &endptr, 10);
-		if (intval > 0x7FFFFFFFUL)
+		m->MoneyValue = 0;
+		intval = strtoll(ptr, &endptr, 10);
+		if (intval > 0x7FFFFFFFFFFFFFFFLL)
```

Comment:
This check shouldn't be necessary anymore since you're putting a signed long long into a signed long long (no change in signing). But you should still somehow handle strings that would overflow